genesis/process/argGenesisBlockCreator.go

```diff
@@ -57,6 +58,7 @@ type runTypeComponentsHandler interface {
 	ShardCoordinatorCreator() sharding.ShardCoordinatorFactory
 	TxPreProcessorCreator() preprocess.TxPreProcessorCreator
 	VMContextCreator() systemSmartContracts.VMContextCreatorHandler
+	VmContainerShardFactoryCreator() factoryVm.VmContainerCreator
```

Comment:
I see we are missing some nil checks for `VmContainerShardFactoryCreator` and `VMContextCreator`.
In `genesisBlockCreator.go` line 218 we need to have the nil checks.
In `genesisBlockCreator_test.go` line 549 we need unit tests.

Reply:
+1

Reply:
Right, good catch, added them